cmake/common/px4_base.cmake

```diff
@@ -915,27 +915,17 @@ endfunction()
 function(px4_create_git_hash_header)
 	px4_parse_function_args(
 		NAME px4_create_git_hash_header
-		ONE_VALUE HEADER
-		REQUIRED HEADER
+		ONE_VALUE OUT
+		REQUIRED OUT
 		ARGN ${ARGN})
-	execute_process(
-		COMMAND git describe --always --tags
-		OUTPUT_VARIABLE git_tag
-		OUTPUT_STRIP_TRAILING_WHITESPACE
-		WORKING_DIRECTORY ${PX4_SOURCE_DIR}
-		)
-	message(STATUS "GIT_TAG = ${git_tag}")
-	execute_process(
-		COMMAND git rev-parse --verify HEAD
-		OUTPUT_VARIABLE git_version
-		OUTPUT_STRIP_TRAILING_WHITESPACE
-		WORKING_DIRECTORY ${PX4_SOURCE_DIR}
+	file(WRITE ${OUT} "")
+	add_custom_command(
+		OUTPUT __fake
```

Comment:
the "ALL" option is for the add_custom_target() function, not the add_custom_command().  In the add_custom_target() I used the ALL option.  If I don't do this fake dependency here, then it will consider the command as successful (since we now have a header generated) and won't run again. I just tried with this diff applied and it doesn't work:

```diff
diff --git a/cmake/common/px4_base.cmake b/cmake/common/px4_base.cmake
index 6775c2b..b5bc0d1 100644
--- a/cmake/common/px4_base.cmake
+++ b/cmake/common/px4_base.cmake
@@ -920,12 +920,12 @@ function(px4_create_git_hash_header)
                ARGN ${ARGN})
        file(WRITE ${OUT} "")
        add_custom_command(
-               OUTPUT __fake
+               OUTPUT ${OUT}
                COMMAND ${PYTHON_EXECUTABLE} ${PX4_SOURCE_DIR}/Tools/px_update_git_header.py ${OUT}
                COMMENT "Generating git hash header"
                )
        add_custom_target(ver_gen ALL
-               DEPENDS ${OUT} __fake)
+               DEPENDS ${OUT})
 endfunction()
 
 #=============================================================================
```